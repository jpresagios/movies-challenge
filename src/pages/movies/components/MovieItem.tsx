import {
  Card,
  CardHeader,
  CardContent,
  CardMedia,
  Typography,
} from '@material-ui/core';
import { makeStyles } from '@material-ui/core/styles';
import React from 'react';
import TextTruncate from 'react-text-truncate';
import MovieProps from '../interfaces/MovieProps';

const useStyles = makeStyles(() => ({
  media: {
    height: 0,
    paddingTop: '60.25%', // 16:9
  },
}));

export default function MovieItem({ title, description, url }: MovieProps) {
  const classes = useStyles();

  return (
    <Card>
      <CardHeader title={(
        <TextTruncate
          line={1}
          text={title}
        />
)}
      />
      <CardMedia
        className={classes.media}
        image={url}
        title={title}
      />
      <CardContent>
        <Typography variant="body2" color="textSecondary" component="p">
          {description}
        </Typography>
      </CardContent>
    </Card>
  );
}
