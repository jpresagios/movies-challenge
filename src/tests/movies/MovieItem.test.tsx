import React from 'react';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import MovieItem from '../../pages/movies/components/MovieItem';

interface Text {
  text: string;
}

jest.mock('react-text-truncate', () => function truncate({ text }: Text) {
  return <div>{text}</div>;
});

test('Movie Item renders correctly', () => {
  const tree = renderer
    .create(<MovieItem title="Movie title" url="Url movie title" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('Movie Item render with the title provide as prop', () => {
  const titleMovie = 'Title movie';
  render(<MovieItem title={titleMovie} url="url movie" />);
  const movieItem = screen.getByText(titleMovie);
  expect(movieItem).toBeInTheDocument();
});
