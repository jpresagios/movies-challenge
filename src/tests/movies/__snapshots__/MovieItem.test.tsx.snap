// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie Item renders correctly 1`] = `
<div
  className="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
>
  <div
    className="MuiCardHeader-root"
  >
    <div
      className="MuiCardHeader-content"
    >
      <span
        className="MuiTypography-root MuiCardHeader-title MuiTypography-h5 MuiTypography-displayBlock"
      >
        <div>
          Movie title
        </div>
      </span>
    </div>
  </div>
  <div
    className="MuiCardMedia-root makeStyles-media-1"
    style={
      Object {
        "backgroundImage": "url(\\"Url movie title\\")",
        "backgroundPosition": "top",
        "height": "35vh",
      }
    }
    title="Movie title"
  />
</div>
`;
